### PR TITLE
uart: Add basic RS485 mode support

### PR DESCRIPTION
**Pull request by qianfan-Zhao**
this pr is based on: 
dts: set a 'boolean' type's value to 1, not 'True' #8401
extract_dts_include: support 'default' value feature #8463

@erwango There has a simple patch you can used to test this pr. 

this is a simple test patch on nucleo_f070rb platform, RS485's flow control pin is PA1. I had changed uart console to RS485 mode. RS485's flow control pin pull up when sending chars.
```
diff --git a/boards/arm/nucleo_f070rb/nucleo_f070rb.dts b/boards/arm/nucleo_f070rb/nucleo_f070rb.dts
index 72f3e2a..df3317f 100644
--- a/boards/arm/nucleo_f070rb/nucleo_f070rb.dts
+++ b/boards/arm/nucleo_f070rb/nucleo_f070rb.dts
@@ -43,6 +43,8 @@
 	current-speed = <115200>;
 	pinctrl-0 = <&usart2_pins_a>;
 	pinctrl-names = "default";
+	rs485-enabled-at-boot-time;
+	rs485-rts-delay=<10 20>;
 	status = "ok";
 };
 
diff --git a/boards/arm/nucleo_f070rb/pinmux.c b/boards/arm/nucleo_f070rb/pinmux.c
index 96c77e1..7181f84 100644
--- a/boards/arm/nucleo_f070rb/pinmux.c
+++ b/boards/arm/nucleo_f070rb/pinmux.c
@@ -18,6 +18,7 @@ static const struct pin_config pinconf[] = {
 #ifdef CONFIG_UART_STM32_PORT_2
 	{STM32_PIN_PA2, STM32F0_PINMUX_FUNC_PA2_USART2_TX},
 	{STM32_PIN_PA3, STM32F0_PINMUX_FUNC_PA3_USART2_RX},
+	{STM32_PIN_PA1, STM32F0_PINMUX_FUNC_PA1_USART2_RTS},
 #endif	/* CONFIG_UART_STM32_PORT_1 */
 #ifdef CONFIG_I2C_1
 	{STM32_PIN_PB8, STM32F0_PINMUX_FUNC_PB8_I2C1_SCL},
```

Next is waveform captured by logic analyzer: (the first line is RS485's flow control, the second is usart's tx)
rs4851
rs4851
rs4851

Assert time is 10.3us
Deassert time is: 17.1us

### CODE REVIEW

**Comment by zhaozhongchen** [comment]
@erwango @ydamigos please have a look

File: arch/arm/soc/st_stm32/stm32f0/dts.fixup [hunk 1]
```diff
@@ -85,4 +85,46 @@
 #define CONFIG_PWM_STM32_17_DEV_NAME            ST_STM32_PWM_40014800_PWM_LABEL
 #define CONFIG_PWM_STM32_17_PRESCALER           ST_STM32_PWM_40014800_PWM_ST_PRESCALER
 
+#ifdef ST_STM32_USART_40013800_RS485_ENABLED_AT_BOOT_TIME
+	#define CONFIG_UART_STM32_USART_1_RS485_ENABLED			1
```
Comment (ulfalizer) [inline arch/arm/soc/st_stm32/stm32f0/dts.fixup]: Might want to use some naming other than `CONFIG_...`, to make it clear that these aren't coming from Kconfig.

That might help with finding references to undefined Kconfig symbols later too.

Comment (zhaozhongchen) [reply arch/arm/soc/st_stm32/stm32f0/dts.fixup]: Seems most items in dts.fixup for stm32 series has a CONFIG_,.,, prefix, thats was aren't coming from Kconfig both. I prefer open another PR to remove the CONFIG_ prefix, not is this PR.

What's yours opinion?

File: arch/arm/soc/st_stm32/stm32f0/dts.fixup [hunk 2]
```diff
@@ -85,4 +85,46 @@
...
+#if !defined(ST_STM32_USART_40013800_RS485_RTS_ACTIVE_LOW)
+	#define CONFIG_UART_STM32_USART_1_RS485_RTS_ACTIVE_LOW		0
```
Comment (ydamigos) [inline arch/arm/soc/st_stm32/stm32f0/dts.fixup]: The fixup file isn't the best place to handle optional properties from DT.

**Review by ydamigos — CHANGES_REQUESTED** [review]
The driver should handle the optional properties from DT. Keep in mind that the dts.fixup files will be removed in the future.

**Comment by zhaozhongchen** [comment]
@ydamigos 

> The fixup file isn't the best place to handle optional properties from DT.

Yes I know but I can't find a better way.

rs485-enabled-at-boot-time is a 'boolean' type, if it is exist in dts, there has sometings like this:
```
#define ST_STM32_USART_40004400_RS485_ENABLED_AT_BOOT_TIME		True
```

But if you haven't set rs485-enabled-at-boot-time, DTS generater doesn't define 'ST_STM32_USART_40004400_RS485_ENABLED_AT_BOOT_TIME'. So I need handler thats by myslef.

As the same rs485-rts-delay is an array if not defined we need keep the default value <0 0>, but the DTS generater doesn't has a default value feature.

So I need a way to handler this, I don't want handle on drivers, so I just let them in dts.fixup.
Is there has another better way to handle this?

File: arch/arm/soc/st_stm32/stm32f0/dts.fixup (hunk 1 above)
Comment (ulfalizer) [reply arch/arm/soc/st_stm32/stm32f0/dts.fixup]: Fine by me, if you'd need to change a bunch of unrelated stuff.

Comment (zhaozhongchen) [reply arch/arm/soc/st_stm32/stm32f0/dts.fixup]: can you write an issue?

Comment (ulfalizer) [reply arch/arm/soc/st_stm32/stm32f0/dts.fixup]: Hmm... not sure how I'd phrase it. Might be other places with non-Kconfig `CONFIG_*` macros as well.

If we add some script for detecting references to undefined Kconfig symbols in source files later, all of them would probably have to be fixed at that point at least.

**Comment by zhaozhongchen** [comment]
@erwango @ydamigos I had updated, please recheck

**Review by erwango — CHANGES_REQUESTED** [review]
Please provide a sample so we can see how this all interacts and one can actually test.

**Pull request closed without merging**